### PR TITLE
Add support for logrotate in sairedis

### PR DESCRIPTION
**Pull request by kcudnik**

### CODE REVIEW

File: lib/src/sai_redis_record.cpp [hunk 1]
```diff
@@ -22,10 +22,34 @@ std::string getTimestamp()
...
 // recording needs to be enabled explicitly
 volatile bool g_record = false;
+volatile bool g_logrotate = false;
 
 std::ofstream recording;
 std::mutex g_recordMutex;
 
+std::string recfile = "dummy.rec";
+
+void logRotate()
```
Comment (qiluo-msft) [inline lib/src/sai_redis_record.cpp]: >logRotate [](start = 5, length = 9)

'Rotate' here is misleading. It just releases and fetches the file handle.

File: lib/src/sai_redis_record.cpp [hunk 2]
```diff
@@ -22,10 +22,34 @@ std::string getTimestamp()
...
+
+    recording.close();
+
+    /*
+     * On log rotate we will use the same file name, we are assuming that
+     * logrotate deamon move filename to filename.1 and we will create new
+     * empty file here.
+     */
+
+    recording.open(recfile);
```
Comment (qiluo-msft) [inline lib/src/sai_redis_record.cpp]: >open [](start = 14, length = 4)

For performance concern, refer [https://stackoverflow.com/a/164770/2514803](https://stackoverflow.com/a/164770/2514803). Just FYI.

File: lib/src/sai_redis_record.cpp (hunk 1 above)
Comment (kcudnik) [inline lib/src/sai_redis_record.cpp]: can you suggest other name ?

Comment (qiluo-msft) [inline lib/src/sai_redis_record.cpp]: How about 'logShift‘？

Comment (kcudnik) [inline lib/src/sai_redis_record.cpp]: sure

File: lib/src/sai_redis_record.cpp (hunk 2 above)
Comment (lguohan) [inline lib/src/sai_redis_record.cpp]: this is periodically, perhaps once per several hours. not sure about the performance concern here.

File: lib/src/sai_redis_record.cpp (hunk 1 above)
Comment (lguohan) [inline lib/src/sai_redis_record.cpp]: logfileRelease

Comment (lguohan) [inline lib/src/sai_redis_record.cpp]: or logfileReopen

Comment (kcudnik) [inline lib/src/sai_redis_record.cpp]: done

**Comment by qiluo-msft** [comment]
Just throw the question without a quick solution: Is it possible not to add new sairedis interface and be transparent to application?

**Comment by lguohan** [comment]
the interface is provided to send reopen the log file, need an api to do it.

**Comment by lguohan** [comment]
:shipit:

**Comment by kcudnik** [comment]
i think thats the point to have our specific apis that can influence sairedis over standard sai interface

**Pull request merged**